### PR TITLE
Update documentation to reflect removal of default conversions

### PR DESCRIPTION
**Pull request by dacodas**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

This touches up the documentation for apimachinery/pkg/conversion to clarify that the default golang copy mechanisms are not called or supported.

#### Which issue(s) this PR fixes:

Reading the Develoment Guide and Your First Contribution documents, I was unsure if I needed to file an issue before submitting a pull request. I will, of course, file an issue if required.

#### Special notes for your reviewer:

A look at this [history of this file][history] and the commits referenced therein clearly show that this feature was disabled. The documentation simply has to be made current.

Our team was investigating this package given the documented behavior of supporting golang's "default copying mechanism". This change in documentation may save future investigators some time in the future 🕵️

[history]: https://github.com/kubernetes/kubernetes/commits/master/staging/src/k8s.io/apimachinery/pkg/conversion/converter.go

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dacodas** [comment]
Alright, I have signed the CLA messieurs

File: staging/src/k8s.io/apimachinery/pkg/conversion/converter.go [hunk 1]
```diff
@@ -190,10 +190,9 @@ func (c *Converter) RegisterIgnoredConversion(from, to interface{}) error {
 }
 
 // Convert will translate src to dest if it knows how. Both must be pointers.
-// If no conversion func is registered and the default copying mechanism
-// doesn't work on this type pair, an error will be returned.
-// 'meta' is given to allow you to pass information to conversion functions,
-// it is not used by Convert() other than storing it in the scope.
+// If no conversion func is registered for this type pair an error will be
```
Comment (yliaog) [inline staging/src/k8s.io/apimachinery/pkg/conversion/converter.go:193]: nit, add a "," at the end of If clause, i.e., "If no conversion func is registered for this type pair, an error will be"

**Comment by dacodas** [comment]
Oh my, I have spent my whole life ignoring commas in this situation. Now that I look it up, I see that various style guides indicate its use here.

I should have paid more attention in school 🤣😋

Thanks! Making the change now

**Comment by dacodas** [comment]
Alright, sorry y'all. Unsure about the CLA Authorization why it's still giving issues.

When I look at this commit, I see no mention of my work e-mail, and yet the CLA Authorization failure is suggesting to me that I need to approve the CLA from my work e-mail. 

Is there any work around for this, or am I misunderstanding something, by any chance?

```
kubernetes $ git cat-file -p 5f975b2b5c405ea6722c4b24cb7d196ca31f4086
tree 7176b46ef5cfa6c6e1111d5a9cd935f4bb25ada7
parent ce52600a3af533b155a200e2c8a494cd6fc45ad5
author Dacoda Strack <dacoda.strack@gmail.com> 1643577103 -0800
committer Dacoda Strack <dacoda.strack@gmail.com> 1643759688 -0800

Update documentation to reflect removal of default conversions
```

Sorry for the hassle after you gave the /lgtm

**Comment by dacodas** [comment]
Alright, my apologies, I re-signed the CLA. I was under the (perhaps mistaken) impression that I only had to sign once and that it would let me continue from there.

Thank you for your patience <3 I think it still is asking for a /lgtm after the previous label was removed

**Pull request closed without merging**